lib/block_view.go

```diff
@@ -6768,6 +6821,21 @@ func (bav *UtxoView) _connectCreateNFT(
 	if postEntry.IsNFT {
 		return 0, 0, nil, RuleErrorCreateNFTOnPostThatAlreadyIsNFT
 	}
+	if isBuyNow && blockHeight < BuyNowNFTBlockHeight {
+		return 0, 0, nil, errors.Wrapf(RuleErrorBuyNowNFTBeforeBlockHeight, "_connectCreateNFT: ")
+	}
+	// We only need to make these checks if we are past the BuyNowNFTBlockHeight
+	if blockHeight >= BuyNowNFTBlockHeight {
```

Comment:
Similar to the _connectUpdateNFT() comment, it seems better to reduce complexity and remove this if statement / always run these checks.